airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -796,7 +796,11 @@ def test_connection(self):
         """
         try:
             session = self.get_session()
-            conn_info = session.client("sts").get_caller_identity()
+            test_endpoint_url = self.conn_config.extra_config.get("test_endpoint_url")
+            conn_info = session.client(
+                "sts",
+                endpoint_url=test_endpoint_url,
```

Comment:
Did you tested it when `test_endpoint_url ` is None. My experience with `boto3` is they make a difference between a parameter value being `None` and a parameter not provided to the function

Reply:
Both of them returned successful result when I've tested both being None and a parameter not provided to the function.

I've executed code below
```python
conn = Connection(
    conn_id="sample_aws_connection",
    conn_type="aws",
    login="access_key",  # Reference to AWS Access Key ID
    password="secret_key",  # Reference to AWS Secret Access Key
    extra={
        "test_endpoint_url": "https://sts.us-east-1.amazonaws.com",
        "region_name": "us-east-1"
    },
)

env_key = f"AIRFLOW_CONN_{conn.conn_id.upper()}"
conn_uri = conn.get_uri()
os.environ[env_key] = conn_uri
conn.test_connection()
```

and I've compared results of two connection. Both of them returned successful result, and there's difference of ```RequestID``` in ```ResponseMetadata``` and ```x-amzn-requestid```, ```date``` in ```HTTPHeaders``` 
```python
conn_info = session.client(
	"sts",
	endpoint_url=test_endpoint_url,
	).get_caller_identity()

conn_info_without_endpoint_url = session.client(
	"sts"
	).get_caller_identity()
```

Is there anything more needed to be checked?

Reply:
Oops there's difference in endpoint of sts session. When I've printed endpoint, former returns url of ```test_endpoint_url``` and later returns Global endpoint.

Reply:
In the code you provided, you dont test when `endpoint_url` is `None` but I assume you tested it as you said. LGTM then